datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -233,11 +233,11 @@ impl BatchPartitioner {
     ///
     /// The time spent repartitioning, not including time spent in `f` will be recorded
     /// to the [`metrics::Time`] provided on construction
-    pub fn partition<F>(&mut self, batch: RecordBatch, mut f: F) -> Result<()>
+    pub fn partition<F>(&mut self, batches: Vec<RecordBatch>, mut f: F) -> Result<()>
```

Comment:
Since this is a *pub** API, it would be best if we could keep the old API, mark it as deprecated, and add a new API with the changed arguments

Reply:
Good one